FrontEnd/src/UserAuthentication/FirestoreDB.js

```diff
@@ -1,16 +1,136 @@
-import {database} from './FirebaseConfig';
-import {doc, setDoc} from 'firebase/firestore';
+import { database, storage } from "./FirebaseConfig";
+import {
+  query,
+  where,
+  collection,
+  doc,
+  getDocs,
+  setDoc,
+  getDoc,
+  onSnapshot
+} from "firebase/firestore";
+import { getDownloadURL, ref, uploadBytes } from "firebase/storage";
+import { v4 } from "uuid";
 
-async function addUser(userCredential, firstName, lastName, userName, signUpEmail){
-    const uid = userCredential.user.uid;
+async function addUser(userID, firstName, lastName, userName, signUpEmail) {
+  const userDocRef = doc(database, "users", userID);
+  return await setDoc(userDocRef, {
+    userID: userID,
+    FirstName: firstName,
+    LastName: lastName,
+    UserName: userName,
+    Email: signUpEmail,
+  });
+}
+
+async function isDatabaseExist() {
+  const usersCollection = query(collection(database, "users"));
```

Comment:
Can we save "users" as a const like `const DATABASE_USERS_KEY = "users";`

It's used in multiple places in this file.